### PR TITLE
Add GitHub Actions test workflow

### PR DESCRIPTION
**Pull request by ajalt**
This will run the test suite any time code is pushed to a PR, which will allow you to confirm that the PR isn't introducing and regressions.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Test
+
+on:
+  pull_request:
+    paths-ignore:
+      - '*.md'
```
Comment (ajalt) [inline .github/workflows/test.yml:6]: This will skip running the test suite if the only files changes are top level markdown files like the README.

**Comment by valich** [comment]
Cool, thanks! 

JFYI currently there is a testing configuration set up [on TeamCity](https://teamcity.jetbrains.com/viewType.html?buildTypeId=IntelliJMarkdownParser_BuildForIdeaPlugin&guest=1). Unfortunately they do not show up in GH UI, so your PR makes sense anyway. I need to read a little about GH actions, haven't used them before :) I'll get back here soon

**Comment by valich** [comment]
Let's merge it!

As I get it, there are a couple of things to consider in the future:
1. One could use better failed test reporting (although the tests are quick run, it might be useful to look at older failures and see what was fixed)
2. We need to agree on reporting spec failures (from #54) as failures won't be regressions for some time.

**Comment by valich** [comment]
I could not find a way to ignore runs so far, found this: https://github.com/actions/toolkit/issues/399

**Pull request merged**